### PR TITLE
Ps 5.6 myr merge november2016

### PR DESCRIPTION
**Pull request by george-lorch**
Merge upstream November 2016 commits from 35ad59416fee5d840ab2e9144cfc30e70306f64e through 0151b45ae15087e4339ee8214cd52ed1b545d5ac

### CODE REVIEW

File: mysql-test/suite/rocksdb.sys_vars/t/rocksdb_tmpdir_basic.test [hunk 1]
```diff
@@ -0,0 +1,38 @@
+--source include/have_rocksdb.inc
```
Comment (george-lorch) [inline mysql-test/suite/rocksdb.sys_vars/t/rocksdb_tmpdir_basic.test:1]: Test only needs MyRocks, not as default.

File: mysql-test/suite/rocksdb/t/autoinc_vars_thread.test [hunk 2]
```diff
@@ -0,0 +1,53 @@
+--source include/have_rocksdb.inc
```
Comment (george-lorch) [inline mysql-test/suite/rocksdb/t/autoinc_vars_thread.test:1]: Test only needs MyRocks, not as default, properly uses ENGINE=ROCKSDB where necessary.

File: mysql-test/suite/rocksdb/t/autoinc_vars_thread_2.test [hunk 3]
```diff
@@ -0,0 +1,141 @@
+--source include/have_rocksdb.inc
```
Comment (george-lorch) [inline mysql-test/suite/rocksdb/t/autoinc_vars_thread_2.test:1]: Test only needs MyRocks, not as default, properly uses ENGINE=ROCKSDB where necessary.

File: mysql-test/suite/rocksdb/t/commit_in_the_middle_ddl.test [hunk 4]
```diff
@@ -0,0 +1,27 @@
+--source include/have_rocksdb.inc
```
Comment (george-lorch) [inline mysql-test/suite/rocksdb/t/commit_in_the_middle_ddl.test:1]: Test only needs MyRocks, not as default, properly uses ENGINE=ROCKSDB where necessary.

File: mysql-test/suite/rocksdb/t/i_s_trx.test [hunk 5]
```diff
@@ -0,0 +1,18 @@
+--source include/have_rocksdb_as_default.inc
```
Comment (george-lorch) [inline mysql-test/suite/rocksdb/t/i_s_trx.test:1]: Test was renamed on merge in, this should not have been changed to _as_default.inc as the test properly uses ENGINE=ROCKSDB.

File: mysql-test/suite/rocksdb/t/i_s_trx_rpl.test [hunk 6]
```diff
@@ -0,0 +1,42 @@
+--source include/master-slave.inc
```
Comment (george-lorch) [inline mysql-test/suite/rocksdb/t/i_s_trx_rpl.test:1]: Test was renamed on merge in.
missing --source include/have_rocksdb.inc

File: mysql-test/suite/rocksdb/t/rocksdb.test [hunk 7]
```diff
@@ -1924,4 +1924,42 @@ SHOW TABLE STATUS LIKE 't1';
...
+--echo #
+--echo # Issue #411: Setting rocksdb_commit_in_the_middle commits transaction
+--echo # without releasing iterator
+--echo #
+
+CREATE TABLE t1 (id1 bigint(20),
+                 id2 bigint(20),
+                 id3 bigint(20),
+                 PRIMARY KEY (id1, id2, id3))
+                 DEFAULT CHARSET=latin1;
```
Comment (george-lorch) [inline mysql-test/suite/rocksdb/t/rocksdb.test:1936]: missing ENGINE=ROCKSDB

File: mysql-test/suite/rocksdb/t/rocksdb.test [hunk 8]
```diff
@@ -1924,4 +1924,42 @@ SHOW TABLE STATUS LIKE 't1';
...
+
+CREATE TABLE t2 (id1 bigint(20),
+                 id2 bigint(20),
+                 PRIMARY KEY (id1, id2))
+                 DEFAULT CHARSET=latin1;
```
Comment (george-lorch) [inline mysql-test/suite/rocksdb/t/rocksdb.test:1941]: missing ENGINE=ROCKSDB;

File: mysql-test/suite/rocksdb/t/tmpdir.test [hunk 9]
```diff
@@ -0,0 +1,35 @@
+--source include/have_rocksdb.inc
```
Comment (george-lorch) [inline mysql-test/suite/rocksdb/t/tmpdir.test:1]: have_rocksdb is ok

File: storage/rocksdb/ha_rocksdb.cc [hunk 10]
```diff
@@ -837,7 +842,7 @@ static MYSQL_SYSVAR_BOOL(enable_thread_tracking,
 static MYSQL_SYSVAR_LONGLONG(block_cache_size, rocksdb_block_cache_size,
   PLUGIN_VAR_RQCMDARG | PLUGIN_VAR_READONLY,
   "block_cache size for RocksDB",
-  nullptr, nullptr, /* RocksDB's default is 8 MB: */ 8*1024*1024L,
+  nullptr, nullptr, 512*1024*1024L,
```
Comment (george-lorch) [inline storage/rocksdb/ha_rocksdb.cc:845]: Magic number

File: storage/rocksdb/ha_rocksdb.cc [hunk 11]
```diff
@@ -2907,6 +2958,114 @@ class Rdb_snapshot_status : public Rdb_tx_list_walker
...
+                            is_replication,
+                            0, /* skip_trx_api */
+                            tx_impl->is_tx_read_only(),
+                            rdb_trx->IsDeadlockDetect(),
+                            tx_impl->num_ongoing_bulk_load(),
+                            thread_id,
+                            query_str});
+      }
+  }
+};
```
Comment (george-lorch) [inline storage/rocksdb/ha_rocksdb.cc:3042]: Best to put new class in a new header file?

File: storage/rocksdb/rdb_i_s.cc [hunk 12]
```diff
@@ -1080,7 +1080,6 @@ static int rdb_i_s_lock_info_fill_table(
       break;
     }
   }
-
   DBUG_RETURN(ret);
```
Comment (george-lorch) [inline storage/rocksdb/rdb_i_s.cc:1083]: Unnecessary addition of empty line?

File: mysql-test/suite/rocksdb/t/autoinc_vars_thread.test [hunk 13]
```diff
@@ -0,0 +1,53 @@
...
+
+--echo #---------------------------
+--echo # two threads inserting simultaneously with increment > 1
+--echo # Issue #390
+--echo #---------------------------
+
+CREATE TABLE t1 (a INT AUTO_INCREMENT PRIMARY KEY) ENGINE=rocksdb;
+
+# Set up connections
+connect (con1, localhost, root,,);
```
Comment (laurynas-biveinis) [inline mysql-test/suite/rocksdb/t/autoinc_vars_thread.test:12]: --source include/count_sessions.inc

File: mysql-test/suite/rocksdb/t/autoinc_vars_thread.test [hunk 14]
```diff
@@ -0,0 +1,53 @@
...
+
+connection con2;
+reap;
+
+connection default;
+SET debug_sync='RESET';
+
+disconnect con1;
+disconnect con2;
+
```
Comment (laurynas-biveinis) [inline mysql-test/suite/rocksdb/t/autoinc_vars_thread.test:50]: --source include/wait_until_count_sessions.inc

File: mysql-test/suite/rocksdb/t/collation.test [hunk 15]
```diff
@@ -155,7 +155,7 @@ DROP TABLE t2;
 # test invalid regex (missing end bracket)
 --exec echo "" >$MYSQLTEST_VARDIR/log/mysqld.1.err
 SET GLOBAL rocksdb_strict_collation_exceptions="[a-b";
---exec grep -A 1 "Invalid pattern" $MYSQLTEST_VARDIR/log/mysqld.1.err | cut -d] -f2
+--exec grep "Invalid pattern" $MYSQLTEST_VARDIR/log/mysqld.1.err | cut -d] -f2
```
Comment (laurynas-biveinis) [inline mysql-test/suite/rocksdb/t/collation.test:158]: Grepping the master error log file is flaky as its contents are arbitrary due to preceding testcases. The canonical MTR way would be to restart the server with a dedicated testcase error log file (and use one of the mysql-test/include files for the grep, i.e. search_pattern_in_file.inc)

File: mysql-test/suite/rocksdb/t/collation.test [hunk 16]
```diff
@@ -168,7 +168,7 @@ DROP TABLE a, b;
 # test invalid regex (trailing escape)
 --exec echo "" >$MYSQLTEST_VARDIR/log/mysqld.1.err
 SET GLOBAL rocksdb_strict_collation_exceptions="abc\\";
---exec grep -A 1 "Invalid pattern" $MYSQLTEST_VARDIR/log/mysqld.1.err | cut -d] -f2
+--exec grep "Invalid pattern" $MYSQLTEST_VARDIR/log/mysqld.1.err | cut -d] -f2
```
Comment (laurynas-biveinis) [inline mysql-test/suite/rocksdb/t/collation.test:171]: Likewise

File: mysql-test/suite/rocksdb/t/i_s_trx_rpl.test [hunk 17]
```diff
@@ -0,0 +1,42 @@
...
+}
+--enable_query_log
+
+connection slave;
+show variables like 'rocksdb_rpl_skip_tx_api';
+--source include/start_slave.inc
+
+--let $it=0
+--let $stop=0
+while ($stop != 1) {
```
Comment (laurynas-biveinis) [inline mysql-test/suite/rocksdb/t/i_s_trx_rpl.test:25]: I suspect this is going to be unstable without some sort of sync - what happens if slave fully syncs before this loop?

Also this burns CPU while waiting for the condition to become true, I'd add some --sleep 0.1 (and reduce $it limit accordingly)

File: mysql-test/suite/rocksdb/t/tmpdir.test [hunk 18]
```diff
@@ -0,0 +1,35 @@
...
+
+--echo # If rocksdb_tmpdir is NULL or "", temporary file will be created in
+--echo # server configuration variable location(--tmpdir)
+
+CREATE TABLE t1 (a INT PRIMARY KEY, b INT) ENGINE=RocksDB;
+show session variables like 'rocksdb_tmpdir';
+--echo # Connection con1
```
Comment (laurynas-biveinis) [inline mysql-test/suite/rocksdb/t/tmpdir.test:8]: --source include/count_sessions.inc

File: mysql-test/suite/rocksdb/t/tmpdir.test [hunk 19]
```diff
@@ -0,0 +1,35 @@
...
+connect (con2,localhost,root);
+--replace_result $MYSQL_TMP_DIR MYSQL_TMP_DIR
+show session variables like 'rocksdb_tmpdir';
+
+ALTER TABLE t1 ADD INDEX kb(b), ALGORITHM=INPLACE;
+disconnect con2;
+connection default;
+set global rocksdb_tmpdir=NULL;
+DROP TABLE t1;
+
```
Comment (laurynas-biveinis) [inline mysql-test/suite/rocksdb/t/tmpdir.test:34]: --source include/wait_until_count_sessions.inc

**Review by laurynas-biveinis — CHANGES_REQUESTED** [review]

**Comment by george-lorch** [comment]
Thanks again for the review Laurynas, everything recorded and will be sent upstream.

**Pull request merged**